### DISCUSSION TITLE
Smart Step Node

### DISCUSSION THREAD

**Discussion by lanthos** [melMass/comfy_mtb#140] [Q&A] [answered]
Hello!

How exactly does the Smart Step node work?  Do the start and end percent need to add up to 100%?  I think I'm putting step into both Ksamplers (for base and refiner) and end on the first Ksampler and start on the second.  Right now I'm putting 80% in both because I want the first one to stop at 80% and the second one to pick up at 80% but I'm not sure that's correct and my brain objects to putting 160% in there. :D  Any guidance on that would be super helpful, thank you so much!

**Comment by melMass** [comment, marked as answer]
Hey,

So the smart step is just a small utility to use percentage instead of step ratios like the ksampler advanced does.

So for instance let's say you want to only step the second half steps using the ksampler node you would do that, set the total to 20, the first step to 10 the last to 20 right?

![image](https://github.com/melMass/comfy_mtb/assets/7041726/316a2fca-f977-4381-9e59-be5eb39adb62)

To me this is not super practical given that if you change the total steps you must change the start/end accordingly. The same thing using smart step would look like this, 20 steps total, starting at 50% ending at 100%: 

![image](https://github.com/melMass/comfy_mtb/assets/7041726/581990e4-95b6-4d7a-bdfa-ed2a8945054b)

This does exactly the same thing with the added benefit of being able to change the steps only (or percent only, you get the idea)

I hope this clear things up a bit

**Reply by lanthos** [reply]
ahh great, thank you!  That helps a ton.
